### PR TITLE
feat: showing models metadata

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding a Tab in the model detail page for GGUF models.

If the model is already downloaded, using it, otherwise using download url to fetch metadata.

This is possible thanks to the amazing library [@huggingface/gguf](https://www.npmjs.com/package/@huggingface/gguf)

### Screenshot / video of UI

https://github.com/containers/podman-desktop-extension-ai-lab/assets/42176370/c7449021-e428-4202-bbc1-bd693afa0344

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1344

### How to test this PR?

- [x] unit tests has been provided

### CODE REVIEW

File: packages/backend/src/managers/modelsManager.ts [hunk 1]
```diff
@@ -422,4 +424,36 @@ export class ModelsManager implements Disposable {
...
+      if (this.isModelOnDisk(modelId)) {
+        const modelPath = path.normalize(getLocalModelFile(model));
+        console.debug(`[getModelMetadata] reading model ${modelPath}`);
+        result = await gguf(modelPath, { allowLocalFile: true });
+      } else if (model.url) {
+        result = await gguf(model.url);
+      } else {
+        throw new Error('cannot get model metadata');
+      }
+      console.log('result', result);
```
Comment (lstocchi) [inline packages/backend/src/managers/modelsManager.ts:448]: to be removed

File: packages/backend/src/managers/modelsManager.ts [hunk 2]
```diff
@@ -422,4 +424,36 @@ export class ModelsManager implements Disposable {
...
+      return result.metadata;
+    } catch (err: unknown) {
+      data['error'] = err;
+      console.error(err);
+      throw err;
+    } finally {
+      data['duration'] = performance.now() - before;
+      this.telemetry.logUsage('get-metadata');
```
Comment (lstocchi) [inline packages/backend/src/managers/modelsManager.ts:456]:
```suggestion
      this.telemetry.logUsage('get-metadata', data);
```

File: packages/frontend/src/pages/ModelInspect.svelte [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+onMount(() => {
+  return modelsMetadata.subscribe(metadatas => {
+    const records = metadatas[modelId];
+    if (records) {
+      items = Object.entries(records).map(([key, value]) => ({
+        name: key,
+        value: String(value),
+      }));
+      console.log(`found ${items.length} items`);
```
Comment (lstocchi) [inline packages/frontend/src/pages/ModelInspect.svelte:50]: to be removed

File: packages/frontend/src/pages/ModelInspect.svelte [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  }),
+  new TableColumn<Item, string>('value', {
+    renderMapping: item => String(item.value),
+    renderer: TableSimpleColumn,
+  }),
+];
+const row = new TableRow<Item>({});
+
+onMount(() => {
+  return modelsMetadata.subscribe(metadatas => {
```
Comment (lstocchi) [inline packages/frontend/src/pages/ModelInspect.svelte:43]: Why subscribing to a store if the only place where it is written it's here? Would it be simpler to have something like
```
studioClient
        .getModelMetadata(modelId)
        .then(result => {
        items = .....
```

without creating/subscribing to a store which is never updated elsewhere.

**Review by lstocchi — COMMENTED** [review]
I'm a bit puzzled about the UI and some of the values displayed. TBH i'm not able to evaluate which are the values that can be useful but when they are very long they are cut, so not very readable.

E.g.
image

Then i think that using the UI that we use on the k8s resources would present the values in a much more clear way
image

File: packages/frontend/src/pages/ModelInspect.svelte (hunk 4 above)
Comment (axel7083) [reply packages/frontend/src/pages/ModelInspect.svelte:43]: If you leave the page and come back, this is made for caching it

**Comment by axel7083** [comment]
> I'm a bit puzzled about the UI and some of the values displayed. 

I was not sure, I am open to suggestion, but we should keep in mind that those are metadatas. The most useful are probably the one related to the head count, which can be used as suggested by https://github.com/containers/podman-desktop-extension-ai-lab/pull/1280#issuecomment-2196310464

> Then i think that using the UI that we use on the k8s resources would present the values in a much more clear way 

@lstocchi I agree, but that would require to hardcode the keys to decide in advance where to place them, however those information are metadatas, and we are not sure for most of them where they will be. We might want to have it discuss in the UX call what do you think ?

File: packages/frontend/src/pages/ModelInspect.svelte (hunk 4 above)
Comment (lstocchi) [reply packages/frontend/src/pages/ModelInspect.svelte:43]: ah ok, please add a comment as a reminder

**Comment by lstocchi** [comment]
> I was not sure, I am open to suggestion, but we should keep in mind that those are metadatas. The most useful are probably the one related to the head count, which can be used as suggested by [#1280 (comment)](https://github.com/containers/podman-desktop-extension-ai-lab/pull/1280#issuecomment-2196310464)

> @lstocchi I agree, but that would require to hardcode the keys to decide in advance where to place them, however those information are metadatas, and we are not sure for most of them where they will be. We might want to have it discuss in the UX call what do you think ?

Sure. 
Also @slemeur and @jeffmaury can give some suggestions

**Comment by axel7083** [comment]
> > I was not sure, I am open to suggestion, but we should keep in mind that those are metadatas. The most useful are probably the one related to the head count, which can be used as suggested by [#1280 (comment)](https://github.com/containers/podman-desktop-extension-ai-lab/pull/1280#issuecomment-2196310464)
> 
> > @lstocchi I agree, but that would require to hardcode the keys to decide in advance where to place them, however those information are metadatas, and we are not sure for most of them where they will be. We might want to have it discuss in the UX call what do you think ?
> 
> Sure. Also @slemeur and @jeffmaury can give some suggestions

I will convert it to draft then, however this is a requirement for https://github.com/containers/podman-desktop-extension-ai-lab/pull/1280 (which is my critical for this sprint)

**Comment by slemeur** [comment]
I do think the table is not looking consistent with the other screens. 
If we want to quickly move forward on this, I'd recommend the format we have on the cards for the kubecontexts, which would just require to adjust the sizes.
Screenshot 2024-07-09 at 15 44 32

That being said, I agree with @lstocchi and would prefer we take the same approach as for the kube objects.
> but that would require to hardcode the keys to decide in advance where to place them, however those information are metadatas, and we are not sure for most of them where they will be. We might want to have it discuss in the UX call what do you think ?

I think you might take the first word/token of the metadata as the category for each sections. Example `tokenizer.dldld.dldld` would create a section `tokenizer` and then you'd have sub tables. 
The one with only one word/token, would go into "Globals" or something like that.

**Comment by axel7083** [comment]
Closing for now

**Pull request closed without merging**